rules/ensure-clusters-are-created-with-private-nodes/raw.rego

```diff
@@ -0,0 +1,29 @@
+
+package armo_builtins
+
+# fails in case enablePrivateCluster is set to false.
+deny[msga] {
+	obj := input[_]
+	obj.apiVersion == "management.azure.com/v1"
+	obj.kind == "ClusterDescribe"
+	obj.metadata.provider == "aks"
+	config = obj.data
+
+	not isPrivateClusterEnabled(config)
+
+	msga := {
+    	"alertMessage": "Cluster is not private.",
+    	"packagename": "armo_builtins",
+    	"alertScore": 7,
+    	"failedPaths": [""],
+    	"fixPaths":[],
+        "fixCommand": "az aks create --resource-group <private-cluster-resource-group> --name <private-cluster-name> --load-balancer-sku standard --enable-private-cluster --network-plugin azure --vnet-subnet-id <subnet-id> --docker-bridge-address --dns-service-ip --service-cidr",
```

Comment:
Is it possible to enable a cluster as private after it's configured? If so maybe we should change the command to `az aks update`?

Reply:
You have to recreate the cluster in order to have it private, so there's not an `update` command for this hardening setup.